consensus/beacon/consensus.go

```diff
@@ -275,26 +259,21 @@ func (beacon *Beacon) verifyHeaders(chain consensus.ChainHeaderReader, headers [
 // Prepare implements consensus.Engine, initializing the difficulty field of a
 // header to conform to the beacon protocol. The changes are done inline.
 func (beacon *Beacon) Prepare(chain consensus.ChainHeaderReader, header *types.Header) error {
-	beacon.lock.RLock()
-	defer beacon.lock.RUnlock()
 
 	// Transition isn't triggered yet, use the legacy rules for preparation.
-	if !beacon.merger.TDDReached() {
-		return beacon.ethone.Prepare(chain, header)
-	}
-	parent := chain.GetHeader(header.ParentHash, header.Number.Uint64()-1)
-	if parent == nil {
+	reached, err := IsTTDReached(chain, header.ParentHash, header.Number.Uint64()-1)
+	if err != nil {
 		return consensus.ErrUnknownAncestor
```

Comment:
This is a bit weird that you silently swawllow the returned error and convert it to ErrUnknownAncestor. I guess you should return `err` here and then document `IsTTDReached` to only ever return ErrUnknownAncestor